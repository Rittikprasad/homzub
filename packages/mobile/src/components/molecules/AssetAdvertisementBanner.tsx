import React from 'react';
import { StyleSheet, View } from 'react-native';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { AssetAdvertisement, AssetAdvertisementResults } from '@homzhub/common/src/domain/models/AssetAdvertisement';

interface IAssetAdvertisementBannerState {
  banners: AssetAdvertisement;
  activeSlide: number;
}

const ITEM_WIDTH = theme.viewport.width - theme.layout.screenPadding * 2;
export class AssetAdvertisementBanner extends React.PureComponent<{}, IAssetAdvertisementBannerState> {
  public state = {
    banners: {} as AssetAdvertisement,
    activeSlide: 0,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchAdvertisementBanners();
  };

  public render(): React.ReactNode {
    const { banners, activeSlide } = this.state;

    if (banners?.results?.length <= 0 || !banners) {
      return null;
    }

    return (
      <View style={styles.container}>
        <SnapCarousel
          carouselData={banners?.results ?? []}
          carouselItem={this.renderCarouselItem}
          sliderWidth={ITEM_WIDTH}
          itemWidth={ITEM_WIDTH}
          activeIndex={activeSlide}
          containerStyle={styles.radius}
          onSnapToItem={this.onSnapToItem}
          testID="bannerSnap"
        />
        <PaginationComponent
          dotsLength={banners?.results?.length ?? 0}
          activeSlide={activeSlide}
          containerStyle={styles.overlay}
          activeDotStyle={styles.activeDotStyle}
          inactiveDotStyle={styles.inactiveDotStyle}
        />
      </View>
    );
  }

  private renderCarouselItem = (item: AssetAdvertisementResults): React.ReactElement => {
    return <SVGUri uri={item.attachment.link} />;
  };

  public onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  public fetchAdvertisementBanners = async (): Promise<void> => {
    const response: AssetAdvertisement = await DashboardRepository.getAdvertisements({ category: 'general' });
    this.setState({ banners: response });
  };
}

const styles = StyleSheet.create({
  container: {
    marginTop: 16,
    height: 200,
    backgroundColor: theme.colors.white,
  },
  radius: {
    borderRadius: 4,
  },
  overlay: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 0,
    paddingVertical: 10,
  },
  activeDotStyle: {
    borderWidth: 1,
  },
  inactiveDotStyle: {
    backgroundColor: theme.colors.darkTint6,
    borderColor: theme.colors.transparent,
  },
});
