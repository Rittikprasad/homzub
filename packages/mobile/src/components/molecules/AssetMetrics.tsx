import React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, ViewStyle, View } from 'react-native';
import { useSelector } from 'react-redux';
import { theme } from '@homzhub/common/src/styles/theme';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { PricePerUnit } from '@homzhub/common/src/components/atoms/PricePerUnit';

interface IProps {
  header: string;
  colorCode: string;
  selectedAssetType?: string;
  value: string | number;
  isCurrency: boolean;
  cardStyle?: StyleProp<ViewStyle>;
  testID?: string;
  onPressMetrics?: () => void;
  maxLength?: number;
}

const AssetMetrics = (props: IProps): React.ReactElement => {
  const { header, value, cardStyle, testID, onPressMetrics, isCurrency, selectedAssetType, colorCode, maxLength } =
    props;

  const currency = useSelector(UserSelector.getCurrency);

  const handlePress = (): void => {
    if (onPressMetrics) {
      onPressMetrics();
    }
  };

  const renderItem = (): React.ReactElement => {
    return (
      <>
        <Text type="small" textType="semiBold" style={styles.metricName} maxLength={maxLength}>
          {header}
        </Text>
        {isCurrency ? (
          <PricePerUnit
            textStyle={styles.metricName}
            currency={currency}
            priceTransformation={false}
            price={value as number}
          />
        ) : (
          <Text type="large" textType="semiBold" style={styles.metricName}>
            {value}
          </Text>
        )}
      </>
    );
  };

  return (
    <TouchableOpacity onPress={handlePress} testID={testID}>
      <View
        style={[
          styles.container,
          cardStyle,
          { backgroundColor: colorCode },
          header === selectedAssetType && styles.selectedItem,
        ]}
      >
        {renderItem()}
      </View>
    </TouchableOpacity>
  );
};

const memoizedComponent = React.memo(AssetMetrics);
export { memoizedComponent as AssetMetrics };

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    paddingVertical: 8,
    marginVertical: 12,
    marginStart: 12,
  },
  metricName: {
    textAlign: 'center',
    marginVertical: 4,
    color: theme.colors.white,
  },
  selectedItem: {
    borderWidth: 1.25,
    borderColor: theme.colors.white,
    shadowColor: theme.colors.shadow,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.32,
    shadowRadius: 5.46,
    elevation: 9,
  },
});
