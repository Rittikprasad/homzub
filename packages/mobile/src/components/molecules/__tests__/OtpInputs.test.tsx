import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { OtpInputs } from '@homzhub/common/src/components/molecules/OtpInputs';

let props: any;
const mockFunction = jest.fn();

describe.skip('OtpInputs', () => {
  const createTestProps = (testProps: any): object => ({
    bubbleOtp: mockFunction,
    toggleError: mockFunction,
    ...testProps,
  });
  props = createTestProps({});

  it('should match snapshot', () => {
    const wrapper = shallow(<OtpInputs {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for error', () => {
    props = createTestProps({
      error: 'Error',
    });
    const wrapper = shallow(<OtpInputs {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call focus next function ', () => {
    const wrapper = shallow(<OtpInputs {...props} />);
    // @ts-ignore
    wrapper.instance().renderInputs();
    // @ts-ignore
    wrapper.find('[testID="otpInput"]').at(0).prop('onChangeText')();
    // @ts-ignore
    wrapper.find('[testID="otpInput"]').at(0).prop('onChangeText')(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call focus previous function ', () => {
    const wrapper = shallow(<OtpInputs {...props} />);
    // @ts-ignore
    wrapper.instance().renderInputs();
    wrapper
      .find('[testID="otpInput"]')
      .at(0)
      .simulate('keyPress', { nativeEvent: { key: '' } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call otp handler function ', () => {
    const wrapper = shallow(<OtpInputs {...props} />);
    // @ts-ignore
    wrapper.instance().otpHandler('message');
    expect(toJson(wrapper)).toMatchSnapshot();

    // @ts-ignore
    wrapper.instance().otpHandler('123456');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
