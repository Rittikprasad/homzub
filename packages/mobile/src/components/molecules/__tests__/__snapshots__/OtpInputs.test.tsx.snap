// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtpInputs should call focus next function  1`] = `
<View
  style={
    Object {
      "marginVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      key="0-otp"
      pointerEvents="auto"
    >
      <Component
        allowFontScaling={true}
        autoFocus={true}
        caretHidden={true}
        key="0-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value={1}
      />
    </View>
    <View
      key="1-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="1-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="2-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="2-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="3-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="3-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="4-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="4-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="5-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="5-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Label
    style={
      Object {
        "color": "#F23C06",
        "marginTop": 8,
      }
    }
    type="regular"
  >
    Error
  </Label>
</View>
`;

exports[`OtpInputs should call focus previous function  1`] = `
<View
  style={
    Object {
      "marginVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      key="0-otp"
      pointerEvents="auto"
    >
      <Component
        allowFontScaling={true}
        autoFocus={true}
        caretHidden={true}
        key="0-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="1-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="1-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="2-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="2-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="3-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="3-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="4-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="4-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="5-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="5-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Label
    style={
      Object {
        "color": "#F23C06",
        "marginTop": 8,
      }
    }
    type="regular"
  >
    Error
  </Label>
</View>
`;

exports[`OtpInputs should call otp handler function  1`] = `
<View
  style={
    Object {
      "marginVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      key="0-otp"
      pointerEvents="auto"
    >
      <Component
        allowFontScaling={true}
        autoFocus={true}
        caretHidden={true}
        key="0-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="1-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="1-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="2-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="2-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="3-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="3-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="4-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="4-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="5-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="5-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Label
    style={
      Object {
        "color": "#F23C06",
        "marginTop": 8,
      }
    }
    type="regular"
  >
    Error
  </Label>
</View>
`;

exports[`OtpInputs should call otp handler function  2`] = `
<View
  style={
    Object {
      "marginVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      key="0-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={true}
        caretHidden={true}
        key="0-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value="1"
      />
    </View>
    <View
      key="1-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="1-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value="2"
      />
    </View>
    <View
      key="2-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="2-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value="3"
      />
    </View>
    <View
      key="3-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="3-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value="4"
      />
    </View>
    <View
      key="4-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="4-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value="5"
      />
    </View>
    <View
      key="5-otp"
      pointerEvents="auto"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="5-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value="6"
      />
    </View>
  </View>
  <Label
    style={
      Object {
        "color": "#F23C06",
        "marginTop": 8,
      }
    }
    type="regular"
  >
    Error
  </Label>
</View>
`;

exports[`OtpInputs should match snapshot 1`] = `
<View
  style={
    Object {
      "marginVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      key="0-otp"
      pointerEvents="auto"
    >
      <Component
        allowFontScaling={true}
        autoFocus={true}
        caretHidden={true}
        key="0-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#C9D6DF",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="1-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="1-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#C9D6DF",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="2-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="2-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#C9D6DF",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="3-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="3-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#C9D6DF",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="4-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="4-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#C9D6DF",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="5-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="5-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#C9D6DF",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`OtpInputs should match snapshot for error 1`] = `
<View
  style={
    Object {
      "marginVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      key="0-otp"
      pointerEvents="auto"
    >
      <Component
        allowFontScaling={true}
        autoFocus={true}
        caretHidden={true}
        key="0-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="1-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="1-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="2-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="2-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="3-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="3-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="4-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="4-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      key="5-otp"
      pointerEvents="none"
    >
      <Component
        allowFontScaling={true}
        autoFocus={false}
        caretHidden={true}
        key="5-otp"
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        onKeyPress={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#F23C06",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 48,
            "textAlign": "center",
            "width": 48,
          }
        }
        testID="otpInput"
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Label
    style={
      Object {
        "color": "#F23C06",
        "marginTop": 8,
      }
    }
    type="regular"
  >
    Error
  </Label>
</View>
`;
